src/numberparse/correct.rs

```diff
@@ -82,7 +82,10 @@ impl<'de> Deserializer<'de> {
 
             #[cfg(feature = "swar-number-parsing")]
             {
-                if is_made_of_eight_digits_fast(&buf[idx..]) {
+                let slice = &buf[idx..idx + 8];
+                // safety: we have 8 bytes
+                let chars: [u8; 8] = unsafe { slice.try_into().unwrap_unchecked() };
```

Comment:
It must copy the data as a u64 must have the alignmet of u64 and not of a u8. That's what caused the panic.

Reply:
The try_into doesn't copy, that's good. The `read_unaligned` / `from_ne_bytes` will copy (or read into a register) depending on what the compiler chooses probably the 2nd when it's used in computation afterwards :)

Reply:
> The try_into doesn't copy, that's good. The read_unaligned / from_ne_bytes will copy (or read into a register) depending on what the compiler chooses probably the 2nd when it's used in computation afterwards :)

That's as expected right?

Reply:
Ja that's fine, what I'm getting at is that we should probably use:

```rust
unsafe { slice.as_ptr() as *const [u8; 8]; };
```
instead of `try_into` & `unwrap_unchecked`

it creates simpler machine code: https://rust.godbolt.org/z/hcvjT483P